Frends.IMAP.ReadEmail/Frends.IMAP.ReadEmail/IMAP.cs

```diff
@@ -27,10 +29,95 @@ public class IMAP
         /// string Subject.
         /// string BodyText.
         /// string BodyHtml.
+        /// List<string> AttachmentSaveDirs.
         /// }
         /// </returns>
         /// 
 
+        //method generating path to each attachment file with a valid path to enail's directory 
+        public static string GenerateFilePath(MimeEntity attachment, string attachmentsDirectoryPath)
+        {
+            var fileName = "";
+
+            if (attachment is MessagePart)
+            {
+                fileName = attachment.ContentDisposition?.FileName;
+                if (string.IsNullOrEmpty(fileName))
+                    fileName = $"attached-message{Guid.NewGuid()}.eml";
+            }
+            else
+            {
+                var part = (MimePart)attachment;
+                fileName = part.FileName;
+            }
+            return $"{attachmentsDirectoryPath}/{fileName}";
+        }
+
+        public static List<string> SaveMessageAttachments(string directory, bool createDir, MimeMessage message)
+        {
+            var result = new List<string>();
+
+            //if no attachments, just return empty list, avoid creating empty directories
```

Comment:
No need to comment self-evident code.

Reply:
Now there is a comment: "throw an error in case for some reason couldn't create a directory".
Happens only when we're about to create one.